store/orm/orm.go

```diff
@@ -94,174 +76,153 @@ func (orm *ORM) PendingBridgeType(jr models.JobRun) (models.BridgeType, error) {
 	if nextTask == nil {
 		return models.BridgeType{}, errors.New("Cannot find the pending bridge type of a job run with no unfinished tasks")
 	}
-	return orm.FindBridge(nextTask.Task.Type.String())
+	return orm.FindBridge(nextTask.TaskSpec.Type.String())
 }
 
 // FindJob looks up a Job by its ID.
 func (orm *ORM) FindJob(id string) (models.JobSpec, error) {
 	var job models.JobSpec
-	err := orm.One("ID", id, &job)
-	return job, err
+	return job, orm.preloadJobs().First(&job, "id = ?", id).Error
 }
 
 // FindInitiator returns the single initiator defined by the passed ID.
-func (orm *ORM) FindInitiator(ID int) (models.Initiator, error) {
+func (orm *ORM) FindInitiator(ID string) (models.Initiator, error) {
 	initr := models.Initiator{}
-	err := orm.One("ID", ID, &initr)
-	return initr, err
+	return initr, orm.DB.
+		Set("gorm:auto_preload", true).
+		First(&initr, "id = ?", ID).Error
 }
 
-// FindInitiatorsForJob returns all initiators for a specific job.
-func (orm *ORM) FindInitiatorsForJob(jobID string) ([]models.Initiator, error) {
-	initrs := []models.Initiator{}
-	err := orm.Where("JobID", jobID, &initrs)
-	return initrs, err
+func (orm *ORM) preloadJobs() *gorm.DB {
+	return orm.DB.
+		Preload("Initiators", func(db *gorm.DB) *gorm.DB {
+			return db.Order("\"order\" asc")
+		}).
+		Preload("Tasks", func(db *gorm.DB) *gorm.DB {
+			return db.Order("id asc")
+		})
+}
+
+func (orm *ORM) preloadJobRuns() *gorm.DB {
+	return orm.DB.
+		Preload("Initiator").
+		Preload("Overrides").
+		Preload("Result").
+		Preload("TaskRuns", func(db *gorm.DB) *gorm.DB {
+			return db.Set("gorm:auto_preload", true).Order("task_spec_id asc")
```

Comment:
Looks like gorm handles saving the TaskSpecs for us. Do we have any guarantee that they are saved in the order we expect, specifically that the IDs are assigned in the order we expect? Seems like an assumption worth checking.

Reply:
To recap, can we trust gorm to save TaskSpecs in the same order as the array in memory? I believe we can. But to further reassure you, we already have quite a few tests that rely on proper ordering of tasks, and are already testing that assumption.